packages/@ama-sdk/core/src/plugins/timeout/timeout.fetch.ts

```diff
@@ -8,41 +37,52 @@ export class TimeoutFetch implements FetchPlugin {
 
   /** Fetch timeout (in millisecond) */
   public timeout: number;
+  private pauseTimeout$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
 
   /**
    * Timeout Fetch plugin.
    *
    * @param timeout Timeout in millisecond
    */
-  constructor(timeout = 60000) {
+  constructor(timeout = 60000, pauseEventFactory: () => Observable<boolean> = impervaPauseFactory) {
```

Comment:
More permissive to support `() => boolean | Promise<boolean> | Observable<boolean>`